Add global data to the documentation

kanishkbh opened the pull request:
Add global data to the documentation (under Couple Your Code / Advanced Topics).
Ref https://github.com/precice/precice/pull/1549

kanishkbh committed:
Add global data documentation.

kanishkbh committed:
Minor changes.

kanishkbh committed:
Add global data in other parts of the webpage

uekerman commented on _includes/xmlreference.md:
This file is auto-generated. No need to modify here.

uekerman commented on _data/sidebars/docs_sidebar.yml:
```diff
@@ -478,6 +478,10 @@ entries:
           url: /couple-your-code-waveform.html
           output: web, pdf
 
+        - title: Global data exchange
```
```suggestion
        - title: Global data
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,74 @@
+{% endwarning %}
+
+{% note %}
+This feature is only available for two-participant coupling and does not yet support [Multi coupling](https://precice.org/configuration-coupling-multi.html). Further, it does not yet allow [Convergence Measures](https://precice.org/configuration-coupling.html#implicit-coupling-schemes) and [Acceleration](https://precice.org/configuration-acceleration) to be defined for global data. An extension t ...
```
```suggestion
This feature is only available for coupling between two participants, i.e. does not yet support [multi coupling](https://precice.org/configuration-coupling-multi.html). Further, it does not yet allow [convergence measures](https://precice.org/configuration-coupling.html#implicit-coupling-schemes) and [acceleration](https://precice.org/configuration-acceleration) to be defined for global data. An extension to these features is planned.
```
We could eventually link to the follow-up issue here.

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,74 @@
+This feature is only available for two-participant coupling and does not yet support [Multi coupling](https://precice.org/configuration-coupling-multi.html). Further, it does not yet allow [Convergence Measures](https://precice.org/configuration-coupling.html#implicit-coupling-schemes) and [Acceleration](https://precice.org/configuration-acceleration) to be defined for global data. An extension t ...
+{% end note %}
+
+preCICE allows participants to exchange data that is not associated with any mesh. Examples of such data are global background pressure (for e.g. in fluid-acoustic coupling) and angles between coordinate systems (for e.g. in CAMRAD-fluid coupling).
```
Can we explain this without pointing to CAMRAD?

kanishkbh replied:
sure, I just picked it up from the description in https://github.com/precice/precice/issues/202.

How about: "for e.g. in fluid-structure coupling for rotor blades"

uekerman replied:
good enough :+1:

language-wise, I would use 

> (e.g. fluid-structure coupling for rotor blades)

so, no "for" or "in"

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
````diff
@@ -0,0 +1,74 @@
+...
+```
+
+Rest of the global data configuration steps are similar to the usual 'mesh-associated' data as described in [Introduction to configuration](https://precice.org/configuration-introduction.html).
````
```suggestion
Rest of the global data configuration steps are similar to the usual *mesh-associated* data as described in [introduction to configuration](https://precice.org/configuration-introduction.html).
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,74 @@
+    <read-data name="angles">
+...
+    <exchange data="angles" from="SolverOne" to="SolverTwo" />
+...
```
Let's add a rudimentary coupling scheme here. Otherwise, it looks like `exchange` belongs to `participant`.

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,74 @@
+
+
+
+Since global data is not associated with any mesh, it should not be configured with the `use-data` tag under the `mesh` definition. Similarly, all tags like `read-data`, `write-data` and `exchange` that configure global data should be written without `mesh=...`.
```
```suggestion
Since global data is not associated with any mesh, it should not be configured with the `use-data` tag under the `mesh` definition. Similarly, tags such as `read-data`, `write-data`, or `exchange` require no `mesh` attribute if used for global data.
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,74 @@
+---
+title: Exchanging Global Data (data not associated to a mesh)
```
```suggestion
title: Global data
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,74 @@
+title: Exchanging Global Data (data not associated to a mesh)
+permalink: couple-your-code-global-data.html
+keywords: api, adapter, data, global
+summary: "You can define and exchange global data (data not associated to a mesh) with preCICE by using specific optional API functions."
```
```suggestion
summary: "Define and exchange global data (data not associated to a mesh) by using specific optional API functions."
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,74 @@
+
+preCICE allows participants to exchange data that is not associated with any mesh. Examples of such data are global background pressure (for e.g. in fluid-acoustic coupling) and angles between coordinate systems (for e.g. in CAMRAD-fluid coupling).
+
+## Configuration of Global Data
```
```suggestion
## Configuration
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,74 @@
+
+Since global data is not associated with any mesh, it should not be configured with the `use-data` tag under the `mesh` definition. Similarly, all tags like `read-data`, `write-data` and `exchange` that configure global data should be written without `mesh=...`.
+
+## Global Data functions in the API
```
```suggestion
## API
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
````diff
@@ -0,0 +1,74 @@
+
+```
+
+## Supported Functionality
````
```suggestion
## Supported functionality
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,74 @@
+
+## Supported Functionality
+
+Global data exchange is supported in both Explicit and Implicit [coupling schemes](https://precice.org/configuration-coupling.html) with Serial as well as Parallel [coupling flows](https://precice.org/couple-your-code-coupling-flow.html#parallel-coupling-schemes).
```
```suggestion
Global data exchange is supported in both explicit and implicit [coupling schemes](https://precice.org/configuration-coupling.html) with serial as well as parallel [coupling flow](https://precice.org/couple-your-code-coupling-flow.html#parallel-coupling-schemes).
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,74 @@
+
+Within implicit coupling, convergence measures and acceleration are not yet supported.
+
+[Data Initialization](https://precice.org/couple-your-code-initializing-coupling-data.html) and [Time Interpolation](https://precice.org/couple-your-code-waveform.html) are also supported on global data.
```
```suggestion
[Data initialization](https://precice.org/couple-your-code-initializing-coupling-data.html) and [time interpolation](https://precice.org/couple-your-code-waveform.html) are also supported for global data.
```

uekerman reviewed (changes requested):
A few tweaks, but looks good already.

kanishkbh committed:
Title: Global data exchange -> Global data

Co-authored-by: Benjamin Uekermann <benjamin.uekermann@gmail.com>

kanishkbh committed:
Undo changes in xmlreference.md
(following review comment https://github.com/precice/precice.github.io/pull/279#discussion_r1255716691)

kanishkbh committed:
Update note about feature limitations

Co-authored-by: Benjamin Uekermann <benjamin.uekermann@gmail.com>

kanishkbh committed:
Update title -> make shorter

Co-authored-by: Benjamin Uekermann <benjamin.uekermann@gmail.com>

kanishkbh committed:
Update page summary: make crisp

Co-authored-by: Benjamin Uekermann <benjamin.uekermann@gmail.com>

kanishkbh committed:
Crisp subheader

Co-authored-by: Benjamin Uekermann <benjamin.uekermann@gmail.com>

kanishkbh committed:
Use sentence case

Co-authored-by: Benjamin Uekermann <benjamin.uekermann@gmail.com>

kanishkbh committed:
Use sentence case

Co-authored-by: Benjamin Uekermann <benjamin.uekermann@gmail.com>

kanishkbh committed:
Use sentence case

Co-authored-by: Benjamin Uekermann <benjamin.uekermann@gmail.com>

kanishkbh committed:
Better wording

Co-authored-by: Benjamin Uekermann <benjamin.uekermann@gmail.com>

kanishkbh committed:
Crisp subheader

Co-authored-by: Benjamin Uekermann <benjamin.uekermann@gmail.com>

kanishkbh committed:
Use sentence case

Co-authored-by: Benjamin Uekermann <benjamin.uekermann@gmail.com>

kanishkbh committed:
Add a rudimentary coupling scheme in the xml example.

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,75 @@
+This feature is only available for coupling between two participants, i.e. does not yet support [multi coupling](https://precice.org/configuration-coupling-multi.html). Further, it does not yet allow [convergence measures](https://precice.org/configuration-coupling.html#implicit-coupling-schemes) and [acceleration](https://precice.org/configuration-acceleration) to be defined for global data. An  ...
+{% end note %}
+
+preCICE allows participants to exchange data that is not associated with any mesh. Examples of such data are global background pressure (for e.g. in fluid-acoustic coupling) and angles between coordinate systems (for e.g. in CAMRAD-fluid coupling).
```
```suggestion
preCICE allows participants to exchange data that is not associated with any mesh. Examples of such data are global background pressure (e.g. fluid-acoustic coupling) and angles between coordinate systems (e.g. fluid-structure coupling for rotor blades).
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
````diff
@@ -0,0 +1,75 @@
+...
+```
+
+
````
```suggestion
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
````diff
@@ -0,0 +1,75 @@
+```
+
+
+
````
```suggestion
```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
````diff
@@ -0,0 +1,75 @@
+...
+```
+
+For SolverTwo:
````
```suggestion
For SolverTwo:

```

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,75 @@
+
+Within implicit coupling, convergence measures and acceleration are not yet supported.
+
+[Data initialization](https://precice.org/couple-your-code-initializing-coupling-data.html) and [time interpolation](https://precice.org/couple-your-code-waveform.html) are also supported for global data.
```
```suggestion
[Data initialization](https://precice.org/couple-your-code-initializing-coupling-data.html) and [time interpolation](https://precice.org/couple-your-code-waveform.html) are also supported for global data.

```

uekerman committed:
Update pages/docs/couple-your-code/couple-your-code-global-data.md

uekerman committed:
Update pages/docs/couple-your-code/couple-your-code-global-data.md

uekerman committed:
Update pages/docs/couple-your-code/couple-your-code-global-data.md

uekerman committed:
Update pages/docs/couple-your-code/couple-your-code-global-data.md

uekerman commented on pages/docs/couple-your-code/couple-your-code-global-data.md:
```diff
@@ -0,0 +1,74 @@
+
+{% note %}
+This feature is only available for coupling between two participants, i.e. does not yet support [multi coupling](https://precice.org/configuration-coupling-multi.html). Further, it does not yet allow [convergence measures](https://precice.org/configuration-coupling.html#implicit-coupling-schemes) and [acceleration](https://precice.org/configuration-acceleration) to be defined for global data. An  ...
+{% end note %}
```
```suggestion
{% endnote %}
```

uekerman committed:
Update pages/docs/couple-your-code/couple-your-code-global-data.md
